app/src/main/kotlin/compiler/common/diagnostics/Diagnostic.kt

```diff
@@ -8,4 +10,22 @@ sealed class Diagnostic {
             TODO("Not yet implemented")
         }
     }
+
+    class ParserError(
+        val symbol: String?,
+        val start: Location,
+        val end: Location,
+        val expectedSymbols: List<String>
```

Comment:
Why store symbols as Strings instead of actual symbols?

Reply:
Because the symbol's type is abstract and in the end we only care about its string representation. On the other hand, sure, it is technically more efficient to _store_ the object itself and only convert it to `String` when needed.

Reply:
Changed to `Any` instead of `String`. The class could be generic on the symbol's type, but we wouldn't gain anything from this.